### PR TITLE
Reorganized navigation

### PR DESCRIPTION
**Pull request by hmhealey**
These changes:
1. This reorganizes the navigation so that once the user is logged in, the root of the navigation stack will be the channel view
2. It moves the ChannelDrawer and RightSideMenu into separate routes from the channel view. I originally had those routes as part of the navigation stack, but that was causing problems with the transition animations so I ended up with something closer to Jarred's code.

### CODE REVIEW

File: app/actions/navigation/index.js [hunk 1]
```diff
@@ -21,57 +29,47 @@ export function goToLogin() {
...
-                }
-            }
+            type: NavigationTypes.NAVIGATION_OPEN_LEFT_DRAWER,
+            route: Routes.ChannelDrawer
         }, getState);
     };
 }
 
-export function goToChannelInfo() {
+export function openRightSideMenu() {
```
Comment (enahum) [inline app/actions/navigation/index.js]: Maybe rename this to `openRightMenuDrawer`

File: app/actions/navigation/index.js [hunk 2]
```diff
@@ -21,57 +29,47 @@ export function goToLogin() {
...
     return async (dispatch, getState) => {
         dispatch({
-            type: NavigationTypes.NAVIGATION_PUSH,
-            route: Routes.ChannelInfo
+            type: NavigationTypes.NAVIGATION_OPEN_RIGHT_DRAWER,
+            route: Routes.RightSideMenu
```
Comment (enahum) [inline app/actions/navigation/index.js]: same with the route

File: app/components/channel_drawer/channel_drawer.js [hunk 3]
```diff
@@ -1,112 +0,0 @@
-// Copyright (c) 2016 Mattermost, Inc. All Rights Reserved.
-// See License.txt for license information.
-
-import React from 'react';
```
Comment (enahum) [inline app/components/channel_drawer/channel_drawer.js]: With the modal PR this is going to be painful yayayayayai

File: app/navigation/router.js [hunk 4]
```diff
@@ -92,13 +105,49 @@ class Router extends React.Component {
...
+        }
+
         return (
-            <NavigationExperimental.Transitioner
-                style={{flex: 1}}
-                navigationState={this.props.navigation}
-                render={this.renderTransition}
-                configureTransition={this.configureTransition}
-            />
+            <Drawer
```
Comment (enahum) [inline app/navigation/router.js]: Very smart to put this here, this seems to me a great decision.

One question: when the stack is showing the drawers there's still going to be a scene in the bottom not completely overlapped right?

File: app/scenes/channel/channel.js [hunk 5]
```diff
@@ -23,6 +20,7 @@ export default class Channel extends React.PureComponent {
             loadProfilesAndTeamMembersForDMSidebar: React.PropTypes.func.isRequired,
             selectInitialChannel: React.PropTypes.func.isRequired,
             openChannelDrawer: React.PropTypes.func.isRequired,
+            openRightSideMenu: React.PropTypes.func.isRequired,
```
Comment (enahum) [inline app/scenes/channel/channel.js]: remember to change the name to drawer

File: app/scenes/channel/channel.js [hunk 6]
```diff
@@ -60,20 +58,16 @@ export default class Channel extends React.PureComponent {
...
-    closeRightSidebar = () => {
-        this.setState({rightSidebarOpen: false});
-    };
-
     openChannelDrawer = () => {
         this.refs.postTextbox.getWrappedInstance().blur();
         this.props.actions.openChannelDrawer();
     }
 
+    openRightSideMenu = () => {
```
Comment (enahum) [inline app/scenes/channel/channel.js]: same here

File: app/scenes/channel_drawer/channel_drawer.js [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+            currentChannel,
+            currentTeam,
+            channels,
+            channelMembers,
+            theme
+        } = this.props;
+
+        return (
+            <View
+                style={{
```
Comment (enahum) [inline app/scenes/channel_drawer/channel_drawer.js]: I prefer to use flex and maybe calculate some of it cause this may hurt on different screen sizes

File: app/scenes/load_team/load_team.js [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+        currentTeam: React.PropTypes.object,
+        actions: React.PropTypes.shape({
+            goToChannelView: React.PropTypes.func.isRequired,
+            handleTeamChange: React.PropTypes.func.isRequired,
+            initWebsocket: React.PropTypes.func.isRequired
+        }).isRequired
+    };
+
+    componentWillMount() {
+        this.props.actions.initWebsocket();
```
Comment (enahum) [inline app/scenes/load_team/load_team.js]: Weren't we doing something different with the WebSocket connection or should we manage that in another PR?

My worry is that if the app is for a time in the background the websocket will disconnect, how is that going to be handled when the app goes back to the foreground.

File: app/scenes/right_side_menu/index.js [hunk 9]
```diff
@@ -0,0 +1,6 @@
+// Copyright (c) 2017 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import RightSideMenuContainer from './right_side_menu_container';
```
Comment (enahum) [inline app/scenes/right_side_menu/index.js]: once more, Drawer

File: app/scenes/right_side_menu/right_side_menu.js [hunk 10]
```diff
@@ -36,43 +37,28 @@ const Styles = StyleSheet.create({
     }
 });
 
-export default class RightSidebarMenu extends React.Component {
+export default class RightSideMenu extends React.Component {
```
Comment (enahum) [inline app/scenes/right_side_menu/right_side_menu.js]: same thing

File: app/actions/navigation/index.js (hunk 1 above)
Comment (hmhealey) [inline app/actions/navigation/index.js]: Makes sense

File: app/components/channel_drawer/channel_drawer.js (hunk 3 above)
Comment (hmhealey) [inline app/components/channel_drawer/channel_drawer.js]: I've already merged/rebased through 3 other PRs so now you get that fun!

File: app/navigation/router.js (hunk 4 above)
Comment (hmhealey) [inline app/navigation/router.js]: Yes, that's how it'll work. We can tweak the props of the drawer to change how much it overlaps

File: app/components/channel_drawer/channel_drawer.js (hunk 3 above)
Comment (enahum) [inline app/components/channel_drawer/channel_drawer.js]: hahaha fair enough I'll do it then

File: app/scenes/channel_drawer/channel_drawer.js (hunk 7 above)
Comment (hmhealey) [inline app/scenes/channel_drawer/channel_drawer.js]: I don't actually know where that came from. I thought I was just leaving it unchanged

File: app/scenes/load_team/load_team.js (hunk 8 above)
Comment (hmhealey) [inline app/scenes/load_team/load_team.js]: Yes, it'll be another PR. I wanted to get the rest of this merged before it conflicts with everything else again.

I think we could actually move the websocket stuff to the ChannelView. Since it'll always be at the root of the stack now, we can have it watch for the app coming out of the background to fire off some stuff.
